tests/lib/prepare.sh

```diff
@@ -404,3 +395,38 @@ prepare_all_snap() {
 
     disable_kernel_rate_limiting
 }
+
+create_snapd_config_classic() {
+    mkdir -p /etc/systemd/system/snapd.service.d
+    cat <<EOF > /etc/systemd/system/snapd.service.d/local.conf
+[Unit]
+StartLimitInterval=0
+[Service]
+Environment=SNAPD_DEBUG_HTTP=7 SNAPD_DEBUG=1 SNAPPY_TESTING=1 SNAPD_CONFIGURE_HOOK_TIMEOUT=30s
+EOF
+    mkdir -p /etc/systemd/system/snapd.socket.d
+    cat <<EOF > /etc/systemd/system/snapd.socket.d/local.conf
+[Unit]
+StartLimitInterval=0
+EOF
+}
+
+restore_each_classic() {
+    # Restore the environment for each service unit
+    systemctl daemon-reload
+    systemctl stop snapd.service snapd.socket
+    find /etc/systemd/system/snapd.service.d -name "*.conf" -delete
+    find /etc/systemd/system/snapd.socket.d -name "*.conf" -delete
+    create_snapd_config_classic
+    if [ -f /etc/environment.bak ]; then
+        cp /etc/environment.bak /etc/environment
+    fi
+    systemctl daemon-reload
+    systemctl start snapd.service snapd.socket
+}
+
+restore_classic() {
+    if [ -f /etc/environment.bak ]; then
+        rm -f /etc/environment.bak
```

Comment:
Agree with the idea, perhaps we could store initial configurations in $SPREAD_PATH/state/ and then restore from there instead of packing this inside the tar for those things are not exclusive of snapd, what do you think?